omnigibson/sensors/vision_sensor.py

```diff
@@ -25,6 +28,16 @@
 sensor_types = sd.SensorType
 
 
+# Duplicate of simulator's render method, used so that this can be done before simulator is created!
+def render():
+    """
+    Refreshes the Isaac Sim app rendering components including UI elements and view ports..etc.
+    """
+    set_carb_setting(og.app._carb_settings, "/app/player/playSimulations", False)
+    og.app.update()
+    set_carb_setting(og.app._carb_settings, "/app/player/playSimulations", True)
```

Comment:
Sorry would you mind adding a more detailed comment explaining why this is needed?

Reply:
yeah -- when OG first starts up, we automatically create / initialize the `Simulator` singleton class. However, mid-initialization, we want to register the "main" viewer camera, which is a `VisionSensor` instance. During the `VisionSensor` initialization, it needs to call `og.sim.render()` in order to propagate some changes, but it can't do so because we're still in the middle of the simulator's initialization process.